File: DSA/LakshaKundra_590013237/Day_38_Question_3.py
def has_redundant_brackets(expression):
    stack = []
    for char in expression:
        if char == ')':
            top = stack.pop()
            flag = True
            while top != '(':
                if top in '+-*/':
                    flag = False
                top = stack.pop()
            if flag:
                return "Yes"
        else:
            stack.append(char)
    return "No"

print(has_redundant_brackets("((a+b))"))
print(has_redundant_brackets("(a+b)"))

File: DSA/NiharikaChand_590017697/Day38Question3.java
import java.util.*;
class Day38Question3 
{
    public static boolean checkRedundantBrk(String expr) 
    {
        Stack<Character> stack = new Stack<>();

        for (char ch : expr.toCharArray()) 
        {
            if (ch == ')') 
            {
                boolean opFound = false;

                while (!stack.isEmpty() && stack.peek() != '(') 
                {
                    char top = stack.pop();
                    if (top == '+' || top == '-' || top == '*' || top == '/') 
                    {
                        opFound = true;
                    }
                }

                if (!stack.isEmpty()) 
                {
                    stack.pop(); 
                }

                if (!opFound) 
                {
                    return true;
                }
            } 
            else 
            {
                stack.push(ch);
            }
        }

        return false;
    }

    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter expression: ");
        String expr = sc.nextLine();

        if (checkRedundantBrk(expr)) 
        {
            System.out.println("Yes");
        } 
        else 
        {
            System.out.println("No");
        }
    }
}

